packages/lodestar/src/network/util.ts

```diff
@@ -50,3 +55,105 @@ export async function initializePeerInfo(peerId: PeerId, multiaddrs: string[]):
   multiaddrs.forEach((ma) => peerInfo.multiaddrs.add(ma));
   return peerInfo;
 }
+
+export function encodeChunkifyResponse(config: IBeaconConfig, method: Method, {err, output}: ChunkResponse): Buffer {
+  if (err) {
+    return encodeResponseError(err);
+  }
+  let data = Buffer.alloc(0);
+  let type: AnyContainerType;
+  switch (method) {
+    case Method.Status:
+      type = config.types.Status;
+      break;
+    case Method.Goodbye:
+      type = config.types.Goodbye;
+      break;
+    case Method.BeaconBlocksByRange:
+    case Method.BeaconBlocksByRoot:
+      type = config.types.SignedBeaconBlock;
+      break;
+    default:
+      throw new Error(`Unhandled method ${method}`);
+  }
+  data = serialize(type, output);
+  return encodeSingleChunk(data);
+}
+
+export function decodeChunkifyResponse(config: IBeaconConfig, method: Method, chunks: Buffer): ResponseBody {
+  switch (method) {
+    case Method.Status:
+    case Method.Goodbye:
+      return decodeSingleChunk(config, method, chunks) as ResponseBody;
+    case Method.BeaconBlocksByRange:
+    case Method.BeaconBlocksByRoot:
+      // lodestar response multple single block chunks
+      // but other clients might response multiple n block chunks
+      // this handle both cases
+      return decodeBlockChunks(config, method, chunks);
+  }
+}
+
+export function encodeBlockChunks(config: IBeaconConfig, blocks: SignedBeaconBlock[]): Buffer {
+  const chunkArr = blocks.map(block => encodeSingleChunk(serialize(config.types.SignedBeaconBlock, block)));
+  return Buffer.concat(chunkArr);
+}
+
+function encodeResponseError(err: Error): Buffer {
+  const b = Buffer.from("c" + "error_message:" + err.message);
+  b[0] = err.message === ERR_INVALID_REQ ? 1 : 2;
+  return b;
+}
+
+function encodeSingleChunk(data: Buffer): Buffer {
+  return Buffer.concat([
+    Buffer.alloc(1),
+    Buffer.from(varint.encode(data.length)),
```

Comment:
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/p2p-interface.md#ssz-encoding-strategy-with-or-without-snappy

So one thing to keep in mind is that the "encoding-dependent" header is dependent on the header (lol).
Here we're hard-coding the varint, which should be ok for now, but should be somehow move the code in a direction where we can abide by different encodings.